cf/query.py

```diff
@@ -788,10 +788,7 @@ def set_condition_units(self, units):
             return
 
         value_units = getattr(value, "Units", None)
-        if value_units is None:
-            # Value has no units
-            value = Data(value, units=units)
-        else:
+        if value_units is not None:
```

Comment:
Not sure about this - it doesn't work if `value` does not have a `Units` attribute: 
```python
>>> q = cf.lt(6)
>>> q._value
6
>>> q.set_condition_units('m')
>>> q._value
6  # No units here!
```
The orginal code does:
```python
>>> q._value
6
>>> q.set_condition_units('m')
>>> q._value
<CF Data(): 6 m>  # Seems right to me ...
```

How was the original code causing problems?

Reply:
Thanks David, I'm updating this in line with our discussion in the video call today which supersedes your comment here but obviously concerns the same line change and underlying issue.